plugins/notifications-backend/src/service/router.ts

```diff
@@ -26,104 +27,121 @@ export async function createRouter(
+      q.read = q.read = stringToBool(q.read);
+
+      getNotificationsCount(dbClient, catalogClient, q)
+        .then(result => res.json(result))
+        .catch(next);
+    },
+  );
+
+  api.register(
+    'setRead',
+    (c, req: express.Request, res: express.Response, next) => {
+      const q: Paths.SetRead.QueryParameters = Object.assign(
+        {},
+        c.request.query,
+      );
 
-    if (Number.isNaN(pageSizeNum) || Number.isNaN(pageNumberNum)) {
-      throw new Error('either pageSize or pageNumber is not a number');
-    }
+      // we need to convert strings to real types due to open PR https://github.com/openapistack/openapi-backend/pull/571
+      q.read = stringToBool(q.read) as boolean;
 
-    const sorting: NotificationsSortingRequest = {
-      fieldName: orderBy?.toString(),
-      direction: orderByDirec?.toString(),
-    };
-
-    getNotifications(
-      dbClient,
-      catalogClient,
-      request.query,
-      pageSizeNum,
-      pageNumberNum,
-      sorting,
-    )
-      .then(notifications => response.json(notifications))
-      .catch(next);
-  });
+      setRead(dbClient, q.messageId, q.user, q.read)
+        .then(result => res.json(result))
+        .catch(next);
+    },
+  );
 
-  router.post('/notifications', (request, response, next) => {
-    createNotification(dbClient, catalogClient, request.body)
-      .then(result => response.json(result))
-      .catch(next);
-  });
+  // create router
 
-  router.put('/notifications/read', (request, response, next) => {
-    const { messageId, user, read } = request.query;
-    if (
-      typeof messageId !== 'string' ||
-      typeof user !== 'string' ||
-      typeof read !== 'string'
-    ) {
-      throw new Error(
-        'the following query parameters must be provided: messageId - string, user - string, read - false/true (boolean)',
-      );
+  const router = Router();
+  router.use(express.json());
+  router.use((req, res, next) => {
+    if (!next) {
+      throw new Error('next is undefined');
     }
 
-    let readBool: boolean;
-
-    switch (read) {
-      case 'true':
-        readBool = true;
-        break;
-      case 'false':
-        readBool = false;
-        break;
-      default:
-        throw new Error(
-          'value of parameter "read" must be either "false" or "true"',
-        );
+    const validation = api.validateRequest(req);
+    if (!validation.valid) {
+      throw validation.errors;
     }
 
-    setRead(dbClient, messageId, user, readBool)
-      .then(() => response.end())
-      .catch(next);
+    api.handleRequest(req, req, res, next);
   });
 
-  router.use(errorHandler());
   return router;
 }
+
+function stringToNumber(s: number | undefined): number | undefined {
+  return s ? Number.parseInt(s.toString(), 10) : undefined;
+}
+
+function stringToBool(s: boolean | undefined): boolean | undefined {
+  if (!s) {
+    return undefined;
+  }
+
+  return s.toString() === 'true' ? true : false;
```

Comment:
maybe it's worth to add `.toLowerCase()`

Reply:
No need.  It comes from the openapi library after validation.  It is just a workaround to convert a LEGAL boolean string to number